### PR TITLE
To formulate the parameters of radio, device , modem

### LINKED ISSUE #92: To formulate the parameters of radio
- [ ] Find the equipment list mapped to the air-interface under test by using the callback RequestForProvidingAcceptanceDataCausesDeterminingTheModemPosition

- [ ] From the equipment uuid in the list , perform the callback RequestForProvidingAcceptanceDataCausesReadingTheRadioComponentIdentifiers.EquipmentInfo

- [ ] If the core-model-1-4:actual-equipment/structure/category == equipment-augment-1-0:EQUIPMENT_CATEGORY_OUTDOOR_UNIT , 
then consider that instance to formulate the following attributes,
      1. equipment-name core-model-1-4:actual-equipment/manufactured-thing/equipment-type/type-name
      2. serial-number core-model-1-4:actual-equipment/manufactured-thing/equipment-instance/serial-number
      3. part-number core-model-1-4:actual-equipment/manufactured-thing/equipment-type/part-type-identifier

### LINKED ISSUE #93: To formulate the parameters of modem
- [ ] Find the equipment list mapped to the air-interface under test by using the callback RequestForProvidingAcceptanceDataCausesDeterminingTheModemPosition

- [ ] From the equipment uuid in the list , perform the callback RequestForProvidingAcceptanceDataCausesReadingTheRadioComponentIdentifiers.EquipmentInfo

- [ ] If the core-model-1-4:actual-equipment/structure/category == equipment-augment-1-0:EQUIPMENT_CATEGORY_MODEM , 
then consider that instance to formulate the following attributes,
      1. equipment-name core-model-1-4:actual-equipment/manufactured-thing/equipment-type/type-name
      2. serial-number core-model-1-4:actual-equipment/manufactured-thing/equipment-instance/serial-number
      3. part-number core-model-1-4:actual-equipment/manufactured-thing/equipment-type/part-type-identifier

### LINKED ISSUE #94: To formulate the parameters of device
- [ ] Find the equipment list mapped to the air-interface under test by using the callback RequestForProvidingAcceptanceDataCausesDeterminingTheModemPosition

- [ ] From the equipment uuid in the list , Perform the callback RequestForProvidingAcceptanceDataCausesReadingTheRadioComponentIdentifiers.EquipmentInfo

- [ ] If the core-model-1-4:actual-equipment/structure/category == equipment-augment-1-0:EQUIPMENT_CATEGORY_FULL_OUTDOOR_UNIT , then consider that instance to formulate the following attributes,
      1. equipment-name core-model-1-4:actual-equipment/manufactured-thing/equipment-type/type-name
      2. serial-number core-model-1-4:actual-equipment/manufactured-thing/equipment-instance/serial-number
      3. part-number core-model-1-4:actual-equipment/manufactured-thing/equipment-type/part-type-identifier

### PR DESCRIPTION
**Pull request by vanithavalluripalli9**
Fixes #92, Fixes #93, Fixes #94

### CODE REVIEW

File: server/service/individualServices/ReadInventoryData.js [hunk 1]
```diff
@@ -7,97 +7,268 @@ const FIRMWARE = {
...
+  for (let i = 0; i < equipmentInfoList.length; i++) {
+    let manufacturedThing = equipmentInfoList[i][EQUIPMENT.CORE_MODEL_ACTUAL_EQUIPMENT][EQUIPMENT.ACTUAL_EQUIPMENT.MANUFACTURED_THING];
+    let category = equipmentInfoList[i][EQUIPMENT.CORE_MODEL_ACTUAL_EQUIPMENT][EQUIPMENT.ACTUAL_EQUIPMENT.STRUCTURE][EQUIPMENT.ACTUAL_EQUIPMENT.CATEGORY];
+    equipment[EQUIPMENT.ACTUAL_EQUIPMENT.EQUIPMENT_NAME] = manufacturedThing[EQUIPMENT.ACTUAL_EQUIPMENT.EQUIPMENT_TYPE][EQUIPMENT.ACTUAL_EQUIPMENT.TYPE_NAME];
+    equipment[EQUIPMENT.ACTUAL_EQUIPMENT.SERIAL_NUMBER] = manufacturedThing[EQUIPMENT.ACTUAL_EQUIPMENT.EQUIPMENT_INSTANCE][EQUIPMENT.ACTUAL_EQUIPMENT.SERIAL_NUMBER];
+    equipment[EQUIPMENT.ACTUAL_EQUIPMENT.PART_NUMBER] = manufacturedThing[EQUIPMENT.ACTUAL_EQUIPMENT.EQUIPMENT_TYPE][EQUIPMENT.ACTUAL_EQUIPMENT.PART_TYPE_IDENTIFIER];
+    if (category === EQUIPMENT.EQUIPMENT_CATEGORY.MODULE + ":" + EQUIPMENT.EQUIPMENT_CATEGORY.MODEM) {
+      equipmentInfo.modem = equipment;
+    } else if (category === EQUIPMENT.EQUIPMENT_CATEGORY.MODULE + ":" + EQUIPMENT.EQUIPMENT_CATEGORY.FULL_OUTDOOR_UNIT) {
+      equipmentInfo.radio = equipment;
```
Comment (PrathibaJee) [inline server/service/individualServices/ReadInventoryData.js:268]: As per the specification , radio is OUTDOOR_UNIT. Please change condition from FULL_OUTDOOR_UNIT to OUTDOOR_UNIT
image

File: server/service/individualServices/ReadInventoryData.js [hunk 2]
```diff
@@ -7,97 +7,268 @@ const FIRMWARE = {
...
+    } else if (category === EQUIPMENT.EQUIPMENT_CATEGORY.MODULE + ":" + EQUIPMENT.EQUIPMENT_CATEGORY.OUTDOOR_UNIT) {
+      equipmentInfo.device = equipment;
```
Comment (PrathibaJee) [inline server/service/individualServices/ReadInventoryData.js:270]: As per the specification , device is FULL_OUTDOOR_UNIT. Please change condition from OUTDOOR_UNIT to FULL_OUTDOOR_UNIT.
image

File: server/service/individualServices/ReadInventoryData.js [hunk 3]
```diff
@@ -7,97 +7,268 @@ const FIRMWARE = {
...
+      *            ;equipment-instance(serial-number))
+    *****************************************************************************************************/
+    let equipmentUuidList = equipmentUuidResponse.equipmentUuidList;
+    for (let i = 0; i < equipmentUuidList.length; i++) {
+      let equipmentUuid = equipmentUuidList[i];
+      let pathParams = [];
+      pathParams.push(mountName);
+      pathParams.push(equipmentUuid);
+      let consequentOperationClientAndFieldParams = await IndividualServiceUtility.getConsequentOperationClientAndFieldParams(forwardingName, stringName)
+      _traceIndicatorIncrementer = ++_traceIndicatorIncrementer;
```
Comment (PrathibaJee) [inline server/service/individualServices/ReadInventoryData.js:228]: will _traceIndicatorIncrementer++ work ?

**Review by PrathibaJee — CHANGES_REQUESTED** [review]

File: server/service/individualServices/ReadInventoryData.js [hunk 4]
```diff
@@ -7,97 +7,268 @@ const FIRMWARE = {
...
+  }
+
+}
+
 /**
  * This method performs the set of procedure to gather the inventory data
  * @param {String} mountName Identifier of the device at the Controller
- * @param {String} linkId Identifier of the microwave link in the planning
  * @param {String} uuidUnderTest Identifier of the air-interface under test
  * @param {Object} ltpStructure ControlConstruct provided from cache
```
[the remark this answers is not in the document]
Comment (vanithavalluripalli9) [reply server/service/individualServices/ReadInventoryData.js:41]: Thank you for the comment . Changed the sequence.

File: server/service/individualServices/ReadInventoryData.js (hunk 1 above)
Comment (vanithavalluripalli9) [reply server/service/individualServices/ReadInventoryData.js]: Thank you for the comment . Sorry , for the copy paste mistake while removing the hardcoding of variables.Changed it now.

File: server/service/individualServices/ReadInventoryData.js (hunk 2 above)
Comment (vanithavalluripalli9) [reply server/service/individualServices/ReadInventoryData.js:270]: Thank you for the comment . Sorry , for the copy paste mistake while removing the hardcoding of variables.Changed it now.

File: server/service/individualServices/ReadInventoryData.js (hunk 3 above)
Comment (vanithavalluripalli9) [reply server/service/individualServices/ReadInventoryData.js:228]: Thank you for the comment. Intially was testing with the same but traceIndicator didn't get incremented as expected so changed it to  ++_traceIndicatorIncrementer.

File: server/service/individualServices/ReadInventoryData.js (hunk 2 above)
Comment (vanithavalluripalli9) [reply server/service/individualServices/ReadInventoryData.js:270]: Thank you for the comment . Sorry , for the copy paste mistake while removing the hardcoding of variables.Changed it now.

File: server/service/individualServices/ReadInventoryData.js (hunk 3 above)
Comment (vanithavalluripalli9) [reply server/service/individualServices/ReadInventoryData.js]: Thank you for the comment. Changed the code accordingly.

**Review by PrathibaJee — APPROVED** [review]

**Pull request merged**